### PR TITLE
Add io.github.efogdev.mpris-timer

### PR DESCRIPTION
**Pull request by efogdev**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly. 
A timer app (CLI/GUI) with seamless GNOME integration, accomplished by pretending to be a player.
- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/efogdev/mpris-timer

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by EfogDev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by EfogDev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by EfogDev** [comment]
Hmm

image

**Comment by efogdev** [comment]
FYI

image

File: io.github.efogdev.mpris-timer.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  - --socket=fallback-x11
+  - --share=ipc
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.mpris.MediaPlayer2
+  - --talk-name=org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer
+  - --talk-name=org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer-0
+  - --talk-name=org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer-1
+  - --talk-name=org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer-2
+  - --talk-name=org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer-3
+  - --talk-name=org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer-4
```
Comment (bbhtt) [inline io.github.efogdev.mpris-timer.yml:21]: this should be automatically granted

> The default policy for the session bus only allows the application to own its own namespace named by $FLATPAK_ID, subnames of it and org.mpris.MediaPlayer2.$FLATPAK_ID for [MPRIS](https://www.freedesktop.org/wiki/Specifications/mpris-spec/).

https://docs.flatpak.org/en/latest/sandbox-permissions.html#d-bus-access

File: misc/io.github.efogdev.mpris-timer.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	<screenshots>
+		<screenshot type="default">
+			<image>https://github.com/user-attachments/assets/80c40dee-1a2f-4729-8f9b-89e5eeb934b9</image>
+		</screenshot>
+		<screenshot>
+			<image>https://github.com/user-attachments/assets/d4d9445d-0783-4c84-aa9f-eea20ec5e690</image>
+		</screenshot>
+	</screenshots>
+	
+	<developer id="efogdev@gmail.com">
```
Comment (bbhtt) [inline misc/io.github.efogdev.mpris-timer.metainfo.xml:40]:
```suggestion
	<developer id="io.github.efogdev">
```

File: misc/io.github.efogdev.mpris-timer.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		<control>keyboard</control>
+		<control>touch</control>
+		<control>gamepad</control>
+	</supports>
+
+	<description>
+		<p>
+			MPRIS Timer is really keyboard friendly! It should be quite intuitive. Use navigation (arrows, tab, shift+tab) or start inputting numbers right away. 
+			Use either as a GUI app or as a CLI tool: see `mpris-timer -help`.
+		</p>
```
Comment (bbhtt) [inline misc/io.github.efogdev.mpris-timer.metainfo.xml:27]: Can you explain what this app does?

File: misc/io.github.efogdev.mpris-timer.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+	<id>io.github.efogdev.mpris-timer</id>
+	<url type="homepage">https://github.com/EfogDev/mpris-timer</url>
+	
+	<name>MPRIS Timer</name>
+	<summary>A timer app utilizing libadwaita and MPRIS interface for seamless GNOME integration</summary>
```
Comment (bbhtt) [inline misc/io.github.efogdev.mpris-timer.metainfo.xml:7]: violates https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines#not-technical

Comment (bbhtt) [reply misc/io.github.efogdev.mpris-timer.metainfo.xml:7]: you can have this in the description.

**Review by bbhtt — CHANGES_REQUESTED** [review]
Icon, desktop file and metainfo should be upstream https://github.com/EfogDev/mpris-timer

**Comment by bbhtt** [comment]
Buildbot is down right now https://status.flathub.org/

**Comment by bbhtt** [comment]
Also why are you closing the previous PRs? The reviews are lost...

**Comment by efogdev** [comment]
> Also why are you closing the previous PRs? The reviews are lost...

Yea sorry, fucked up a little with those. This one is final

**Comment by efogdev** [comment]
> Icon, desktop file and metainfo should be upstream https://github.com/EfogDev/mpris-timer

`misc` dir
image

**Comment by hfiguiere** [comment]
then why are they in this repository?

File: io.github.efogdev.mpris-timer.yml (hunk 1 above)
Comment (efogdev) [reply io.github.efogdev.mpris-timer.yml:21]: `org.mpris.MediaPlayer2.$FLATPAK_ID` should, but the app might be spawned in multiple instances and it is actually expected to. So I'm not sure how else I can handle this
image

Comment (bbhtt) [reply io.github.efogdev.mpris-timer.yml:21]: make it a sub-bus name of `org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer` instead of adding a `-0` directly eg. `org.mpris.MediaPlayer2.io.github.efogdev.mpris-timer.instance0`

**Comment by efogdev** [comment]
@hfiguiere that's how I've interpreted the instruction. This is in fact just a branch of upstream. 

image

**Comment by bbhtt** [comment]
where does it say to add the metainfo files here? It is clearly written to have them upstream. and if you have them upstream, there is no reason to have duplicate copies.

image

**Comment by hfiguiere** [comment]
> that's how I've interpreted the instruction. This is in fact just a branch of upstream.

I don't see where it says to do that.

**Comment by efogdev** [comment]
No problems, fixed everything

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

File: io.github.efogdev.mpris-timer.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+command: mpris-timer
+
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --filesystem=~/.var/app:create
```
Comment (bbhtt) [inline io.github.efogdev.mpris-timer.yml:14]: why is this needed?

File: io.github.efogdev.mpris-timer.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline io.github.efogdev.mpris-timer.yml:15]: do you need this? If it is gtk4, it should go via the portal.

File: io.github.efogdev.mpris-timer.yml (hunk 5 above)
Comment (efogdev) [reply io.github.efogdev.mpris-timer.yml:14]: The app needs access to some kind of persistent cache dir

Comment (bbhtt) [reply io.github.efogdev.mpris-timer.yml:14]: it already has access to its own cache dir in `~/.var/app/$app-id/cache`.

File: io.github.efogdev.mpris-timer.yml (hunk 6 above)
Comment (efogdev) [reply io.github.efogdev.mpris-timer.yml:15]: The app might be started without GUI, I don't create GTK application in that case as a performance optimization

File: io.github.efogdev.mpris-timer.yml (hunk 5 above)
Comment (efogdev) [reply io.github.efogdev.mpris-timer.yml]: fixed

File: io.github.efogdev.mpris-timer.yml (hunk 6 above)
Comment (efogdev) [reply io.github.efogdev.mpris-timer.yml]: also fixed

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by efogdev** [comment]
Should be good to go

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

File: io.github.efogdev.mpris-timer.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+command: mpris-timer
+
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio 
+  - --share=ipc
```
Comment (bbhtt) [inline io.github.efogdev.mpris-timer.yml:14]:
```suggestion
  - --share=ipc
  - --device=dri
```

dri is needed as these has a GUI

Comment (efogdev) [reply io.github.efogdev.mpris-timer.yml:14]: done

**Comment by bbhtt** [comment]
Seems like some icons are missing?

image

image

image



Also you should set the correct id in mpris so that the icon here works:


image

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by efogdev** [comment]
I don't see the build in https://buildbot.flathub.org/#/pendingbuildrequests

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer

**Comment by efogdev** [comment]
Just checked in a VM, everything is resolved now

**Comment by efogdev** [comment]
bot, build io.github.efogdev.mpris-timer
